# Patch to debug this error in the testing phase:
# "Unable to successfully detect the presence of any -std=c++NN flag. This likely indicates an issue with your compiler."
# The error comes from a tryCompileFlag invocation that hides the actual error from the logs
--- libcxx/utils/libcxx/test/dsl.py.ORIG	2026-07-02 12:30:05.056788275 +0200
+++ libcxx/utils/libcxx/test/dsl.py	2026-07-02 12:45:58.967663630 +0200
@@ -212,9 +212,23 @@
     """
     # fmt: off
     with _makeConfigTest(config) as test:
-        out, err, exitCode, timeoutInfo, _, _ = _executeWithFakeConfig(test, [
-            "%{{cxx}} -xc++ {} -Werror -fsyntax-only %{{flags}} %{{compile_flags}} {}".format(os.devnull, flag)
-        ])
+        out, err, exitCode, timeoutInfo, cmd, _ = _executeWithFakeConfig(
+            test,
+            [
+                "%{{cxx}} -xc++ {} -Werror -fsyntax-only %{{flags}} %{{compile_flags}} {}".format(
+                    os.devnull, flag
+                )
+            ],
+        )
+
+        print("=" * 80)
+        print("COMMAND:", cmd)
+        print("EXIT:", exitCode)
+        print("STDOUT:")
+        print(out)
+        print("STDERR:")
+        print(err)
+
         return exitCode, out, err
     # fmt: on
 
